src/v/security/gssapi.h

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#pragma once
+
+#include "bytes/bytes.h"
+
+#include <gssapi/gssapi.h>
+
+#include <string_view>
+
+namespace security::gss {
+
+struct buffer_view : ::gss_buffer_desc {
```

Comment:
kinda feel like we should encapsulate rather than inherit in cases like this where we'll be interacting with C libraries (e.g. gss_release_buffer). wdyt?

Reply:
Done